addons/mail/static/src/core/common/message_service.js

```diff
@@ -30,6 +30,9 @@ export class MessageService {
         if (convertBrToLineBreak(message.body) === body && attachments.length === 0) {
             return;
         }
+        await Promise.all(message.mentionedChannelPromises).then((fetchedChannels) => {
+            mentionedChannels.add(...fetchedChannels);
```

Comment:
Using recordset `add` is really confusing when the recordset is given as param rather than directly read from a record. Especially when default param value is an array, it will just crash.

Reply:
I understand the problem you mentioned here but I checked the code and the only place calling this function provides the `this.props.composer.mentionedChannels` as the param and it will always be a recordset. So that the default value in the func will never be used actually. In this case, what do you prefer to fix the issue? remove the default array param in the function or something else?